pypsa-to-imac

```diff
@@ -0,0 +1,400 @@
+
+            #Energy :Natural gas
+
+            var1['Generation|Yearly|Heat|Gases|Fossil|Natural Gas'] = 0.001*ct*(n.links_t.p2.filter(like ='gas CHP').filter(like =country)).sum().sum()          
+            var1['Generation|Yearly|Heat|Gases|Fossil|Natural Gas'] += 0.001*ct* (n.links_t.p2.filter(like ='gas boiler').filter(like =country)).sum().sum()
+            
+            var1['Generation|Yearly|Heat|Gases|Fossil|Natural Gas|CCS'] = 0.001*ct*(n.links_t.p2.filter(like ='gas CHP CC').filter(like =country)).sum().sum()            
+            
+            #Energy :Hydrogen
+            
+            var1['Generation|Yearly|Heat|Gases|Hydrogen'] = 0.001*ct*(n.links_t.p2.filter(like ='H2 Fuel Cell').filter(like =country)).sum().sum()   
+            var1['Generation|Yearly|Heat|Gases|Hydrogen'] += 0.001*ct*(n.links_t.p3.filter(like ='Fischer-Tropsch').filter(like =country)).sum().sum()     
+            
+            #Energy :Power to gas (?)
+            """"    
+            var1['Generation|Yearly|Heat|Gases|P2G'] = 0.001*ct*(n.links.efficiency.filter(like ='Sabatier').filter(like =country)*\
+                 n.links_t.p1.filter(like ='Sabatier').filter(like =country)).sum().sum()*var1['Generation|Yearly|Heat|Gases|Fossil|Natural Gas']/\
+                 (var1['Generation|Yearly|Heat|Gases|Fossil|Natural Gas']+var1['Generation|Yearly|Electricity|Gases|Fossil|Natural gas'])                
+            """
+            #Energy : Oil
+ 
+            var1['Generation|Yearly|Heat|Liquids|Fossil'] = 0.001*ct*(n.links_t.p1.filter(like ='oil boiler').filter(like =country)).sum().sum() 
+            
+            #Energy : solar thermal
+    
+            var1['Generation|Yearly|Heat|Solar'] = 0.001*ct*(n.generators_t.p.filter(like ='solar thermal').filter(like =country)).sum().sum() 
+            
+                                                
+            # Hydrogen energy  (MW -> GW)  (?)
+
+            var1['Generation|Yearly|Hydrogen|Electricity'] = 0.001*ct*(n.links_t.p1.filter(like ='H2 Electrolysis').filter(like =country)).sum().sum()                   
+            
+            var1['Generation|Yearly|Hydrogen|Gasses|Fossil|Natural gas'] = 0.001*ct*(n.links_t.p1.filter(like ='SMR').filter(like =country)).sum().sum()
+            
+            var1['Generation|Yearly|Hydrogen|Gasses|Fossil|Natural gas|CCS'] = 0.001*ct*(n.links_t.p1.filter(like ='SMR CC').filter(like =country)).sum().sum()            
+            
+            var1['Generation|Yearly|Hydrogen'] = var1['Generation|Yearly|Hydrogen|Electricity']+\
+            var1['Generation|Yearly|Hydrogen|Gasses|Fossil|Natural gas']
+            
+            # P2G energy (MW -> GW) 
+                  
+            var1['Generation|Yearly|P2G|Electricity'] = 0.001*ct*(n.links_t.p1.filter(like ='helmeth').filter(like =country)).sum().sum()
+                
+                  
+            for v in var.keys():
+                ro=[r for r in ds['D'] if r.value==v][0].row
```

Comment:
Save only one representative decimal point

Reply:
Could be problematic as some numbers are very small, I assumed 3 since we're converting everything from MW in pypsa to GW.